app/code/core/Mage/Core/functions.php

```diff
@@ -165,7 +165,7 @@ function mageCoreErrorHandler($errno, $errstr, $errfile, $errline)
         case E_USER_NOTICE:
             $errorMessage .= 'User Notice';
             break;
-        case E_STRICT:
+        case PHP_VERSION_ID < 80400 && E_STRICT:
```

Comment:
The `case` is based on `switch($errno)`, which is not a bool comparison.

Reply:
I dont get it. We switch to operation based on $errno, with an additional condition. I dont have a better idea right now.

Reply:
https://github.com/sebastianbergmann/phpunit/issues/5956